app/api/ada/lib/adaAddressProcessing.js

```diff
@@ -158,3 +165,21 @@ function _addFetchedAddressesInfo(
 
   return fetchedAddressesInfo.concat(newAddressesInfo);
 }
+
+export function generateAddressBatch(
+  indices: Array<number>,
+  cryptoAccount: RustModule.Wallet.Bip44AccountPublic,
+  type: AddressType,
+): Array<string> {
+  const setting = RustModule.Wallet.BlockchainSettings.from_json({
+    protocol_magic: protocolMagic
+  });
+  return indices.map(i => {
+    const pubKey = cryptoAccount.address_key(
+      type === 'Internal',
+      RustModule.Wallet.AddressKeyIndex.new(i)
+    );
+    const addr = pubKey.bootstrap_era_address(setting);
+    return addr.to_base58();
+  });
+}
```

Comment:
Why `'Internal'` is hardcoded in here? Maybe it should be reflected in the function name

Reply:
Maybe you didn't notice but this is a equality check (`===`)
The first argument to `address_key` is whether or not the address you create is a internal. I check to see if `type === 'Internal'` to convert from the enum to a bool.

Reply:
Ah yes, my mistake. Thank you!